### PR TITLE
Implement `uv help` manually instead of using Clap default

### PR DESCRIPTION
**Pull request by zanieb**
Extends #4772 

Implements `uv help` ourselves so we can do things like #4909 
Adds hints to use `uv help` for more details during short help display.

### CODE REVIEW

File: crates/uv/tests/help.rs [hunk 1]
```diff
@@ -11,18 +11,20 @@ fn help() {
...
 
     Commands:
+      help     Display information about a command
       pip      Resolve and install Python packages
       tool     Run and manage executable Python packages
       python   Manage Python installations
       venv     Create a virtual environment
       cache    Manage the cache
       version  Display uv's version
-      help     Print this message or the help of the given subcommand(s)
```
Comment (zanieb) [inline crates/uv/tests/help.rs:25]: Should move this to the bottom again

File: crates/uv/tests/help.rs [hunk 2]
```diff
@@ -225,9 +230,11 @@ fn help_subcommand() {
     success: true
     exit_code: 0
     ----- stdout -----
+
+    ----- stderr -----
     Manage Python installations
 
-    Usage: uv python [OPTIONS] <COMMAND>
+    Usage: python <COMMAND>
```
Comment (konstin) [inline crates/uv/tests/help.rs:237]: This should keep the uv preview

Comment (konstin) [inline crates/uv/tests/help.rs:237]: This should keep the uv preview

**Review by konstin — APPROVED** [review]
Found a way to make it panic, but i do like the more concise help a lot.


```
$ cargo run -q -- help pip install
thread 'main' panicked at /home/konsti/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.8/src/builder/debug_asserts.rs:214:13:
Command install: Argument or group 'offline' specified in 'conflicts_with*' for 'refresh' does not exist
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

File: crates/uv/tests/help.rs (hunk 2 above)
Comment (zanieb) [reply crates/uv/tests/help.rs:237]: Will investigate. I don't know how Clap renders these things in a parent-aware style.

Comment (zanieb) [reply crates/uv/tests/help.rs:237]: Fixed.

**Comment by zanieb** [comment]
Fixed the panic :)

**Comment by charliermarsh** [comment]
Should these be different?

```
puffin on  zb/cli-help-cmd [$] is 📦 v0.2.23 via 🐍 v3.12.3 via 🦀 v1.79.0
❯ cargo run -- python --help
   Compiling uv-cli v0.0.1 (/Users/crmarsh/workspace/puffin/crates/uv-cli)
   Compiling uv v0.2.23 (/Users/crmarsh/workspace/puffin/crates/uv)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 2.81s
     Running `target/debug/uv python --help`
Manage Python installations

Usage: uv python [OPTIONS] <COMMAND>

Commands:
  list       List the available Python installations
  install    Download and install Python versions
  find       Search for a Python installation
  dir        Show the uv Python installation directory
  uninstall  Uninstall Python versions

Options:
  -q, --quiet                                  Do not print any output
  -v, --verbose...                             Use verbose output
      --color <COLOR_CHOICE>                   Control colors in output [default: auto] [possible values: auto, always, never]
      --native-tls                             Whether to load TLS certificates from the platform's native certificate store [env: UV_NATIVE_TLS=]
      --offline                                Disable network access, relying only on locally cached data and locally available files
      --python-preference <PYTHON_PREFERENCE>  Whether to prefer using Python from uv or on the system [possible values: only-managed, installed, managed, system, only-system]
      --python-fetch <PYTHON_FETCH>            Whether to automatically download Python when required [possible values: automatic, manual]
  -n, --no-cache                               Avoid reading from or writing to the cache [env: UV_NO_CACHE=]
      --cache-dir <CACHE_DIR>                  Path to the cache directory [env: UV_CACHE_DIR=]
      --config-file <CONFIG_FILE>              The path to a `uv.toml` file to use for configuration [env: UV_CONFIG_FILE=]
  -h, --help                                   Print help
  -V, --version                                Print version

Use `uv help python` for more details.
puffin on  zb/cli-help-cmd [$] is 📦 v0.2.23 via 🐍 v3.12.3 via 🦀 v1.79.0 took 3s
❯ cargo run -- help python
   Compiling uv-cli v0.0.1 (/Users/crmarsh/workspace/puffin/crates/uv-cli)
   Compiling uv v0.2.23 (/Users/crmarsh/workspace/puffin/crates/uv)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 2.83s
     Running `target/debug/uv help python`
Manage Python installations

Usage: python <COMMAND>

Commands:
  list       List the available Python installations
  install    Download and install Python versions
  find       Search for a Python installation
  dir        Show the uv Python installation directory
  uninstall  Uninstall Python versions
  help       Print this message or the help of the given subcommand(s)

Options:
  -h, --help
          Print help (see a summary with '-h')
```

**Comment by zanieb** [comment]
Yes they should be different but we should be displaying the long-form documentation for all those options in the second invocation. Will investigate that...

**Comment by zanieb** [comment]
Ah I pushed a fix to #4909 but not here by accident.

**Review by charliermarsh — APPROVED** [review]

**Pull request merged**